tests/common.go

```diff
@@ -2892,104 +2892,23 @@ func CreateCluster(name string, kubeconfigPath string, orgID string, cloud_name
-		backupDriver := Inst().Backup
-		switch provider {
-		case drivers.ProviderAws:
-			log.Infof("Create creds for aws")
-			id := os.Getenv("AWS_ACCESS_KEY_ID")
-			expect(id).NotTo(equal(""),
-				"AWS_ACCESS_KEY_ID Environment variable should not be empty")
-
-			secret := os.Getenv("AWS_SECRET_ACCESS_KEY")
-			expect(secret).NotTo(equal(""),
-				"AWS_SECRET_ACCESS_KEY Environment variable should not be empty")
-
-			credCreateRequest := &api.CloudCredentialCreateRequest{
-				CreateMetadata: &api.CreateMetadata{
-					Name:  name,
-					Uid:   uid,
-					OrgId: orgID,
-				},
-				CloudCredential: &api.CloudCredentialInfo{
-					Type: api.CloudCredentialInfo_AWS,
-					Config: &api.CloudCredentialInfo_AwsConfig{
-						AwsConfig: &api.AWSConfig{
-							AccessKey: id,
-							SecretKey: secret,
-						},
-					},
-				},
-			}
-
-			ctx, err := backup.GetAdminCtxFromSecret()
-			log.FailOnError(err, fmt.Sprintf("Failed to fetch px-central-admin ctx: [%v]", err))
-
-			_, err = backupDriver.CreateCloudCredential(ctx, credCreateRequest)
-			if err != nil && strings.Contains(err.Error(), "already exists") {
-				return
-			}
-			expect(err).NotTo(haveOccurred(),
-				fmt.Sprintf("Failed to create cloud credential [%s] in org [%s]", name, orgID))
-		// TODO: validate CreateCloudCredentialResponse also
-		case drivers.ProviderAzure:
-			log.Infof("Create creds for azure")
-			tenantID, clientID, clientSecret, subscriptionID, accountName, accountKey := GetAzureCredsFromEnv()
-			credCreateRequest := &api.CloudCredentialCreateRequest{
-				CreateMetadata: &api.CreateMetadata{
-					Name:  name,
-					Uid:   uid,
-					OrgId: orgID,
-				},
-				CloudCredential: &api.CloudCredentialInfo{
-					Type: api.CloudCredentialInfo_Azure,
-					Config: &api.CloudCredentialInfo_AzureConfig{
-						AzureConfig: &api.AzureConfig{
-							TenantId:       tenantID,
-							ClientId:       clientID,
-							ClientSecret:   clientSecret,
-							AccountName:    accountName,
-							AccountKey:     accountKey,
-							SubscriptionId: subscriptionID,
-						},
-					},
-				},
-			}
-			ctx, err := backup.GetAdminCtxFromSecret()
-			expect(err).NotTo(haveOccurred(),
-				fmt.Sprintf("Failed to fetch px-central-admin ctx: [%v]",
-					err))
-			_, err = backupDriver.CreateCloudCredential(ctx, credCreateRequest)
-			if err != nil && strings.Contains(err.Error(), "already exists") {
-				return
-			}
-			expect(err).NotTo(haveOccurred(),
-				fmt.Sprintf("Failed to create cloud credential [%s] in org [%s]", name, orgID))
-			// TODO: validate CreateCloudCredentialResponse also
-		}
-	})
-}
-
-// CreateCloudCredential creates cloud credetials
-func CreateCloudCredentialNonAdminUser(provider, name string, uid, orgID string, ctx context1.Context) error {
-	log.Infof("Create credential name %s for org %s provider %s", name, orgID, provider)
-	backupDriver := Inst().Backup
+func CreateCloudCredential(provider, credName string, uid, orgID string, ctx context1.Context) error {
```

Comment:
Since you are refactoring the code, I have a suggestion:
The import is named `context1`, whereas there _is_ no other import by the name of `context`. `context1` refers to the context module in go stdlib, so we can leave it as just `context` or make it `goContext` (which makes more sense as to what the import is).
@mkoppal-px , what do you think?

Reply:
> > > Everything seems fine, **except for the following**:
> > > Wherever you've introduced the new `CreateCloudCredentials`, you were required to `GetAdminCtxFromSecret`. But you've implemented the error handling for that _pretty_ inconsistently across various tests in various files. You've used 2 variants:
> > > 
> > > * variant 1:
> > > 
> > > ```
> > > ctx, err := backup.GetAdminCtxFromSecret()
> > > dash.VerifyFatal(err, nil, "Getting context")
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > 
> > > * variant 2:
> > > 
> > > ```
> > > ctx, err := backup.GetAdminCtxFromSecret()
> > > log.FailOnError(err, "Fetching px-central-admin ctx")
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > I feel that everything must be changed to _variant 2_ for 2 reasons:
> > > 
> > > 1. Most other situations where `GetAdminCtxFromSecret` was invoked, involved `log.FailOnError`
> > > 2. More importantly, as I said once before, `log.FailOnError` is used to check for error and fail on an **intermediary step** (afaik), whereas `dash.VerifyFatal` is used in places where one would technically use "assertions" (`left == right`) (which is the _actual_ test/comparison).
> > 
> > 
> > Although it may not appear consistent overall, retrieving ctx is consistent within the test case. Once we standardize the structure and syntax of the test case, we can then address this issue.
> 
> Let us use `log.FailOnError` everywhere

Done. Used log.FailOnError(err, "Fetching px-central-admin ctx") for GetAdminCtx and log.FailOnError(err, "Fetching non admin ctx") for GetNonAdminCtx.

Reply:
We can take this up later. Not critical as of now.